src/applications/vaos/new-appointment/components/DateTimeSelectPage/index.unit.spec.jsx

```diff
@@ -890,164 +891,372 @@ describe('VAOS Page: DateTimeSelectPage', () => {
+        );
+
+        // Have a selected slot, can move to next screen
+        userEvent.click(screen.getByText(/^Continue/));
+        await waitFor(() => {
+          expect(screen.history.push.called).to.be.true;
+        });
+        screen.history.push.reset();
+
+        // Clicking selected date should unselect date
+        userEvent.click(dayOfMonthButton);
+        userEvent.click(screen.getByText(/^Continue/));
+        await waitFor(() => {
+          expect(screen.getByRole('alert')).to.be.ok;
+        });
+        expect(screen.history.push.called).to.be.false;
+      });
     }
 
-    const secondDateString = formatInTimeZone(
-      secondSlotDate,
-      timezone,
-      'EEEE, MMMM do',
-    );
-    dayOfMonthButton = await screen.findByLabelText(
-      new RegExp(secondDateString, 'i'),
-    );
-    userEvent.click(dayOfMonthButton);
-
-    const expectedSecondSlot = formatInTimeZone(
-      secondSlotDate,
-      timezone,
-      "h:mm a 'option selected'",
-    );
-    userEvent.click(
-      await screen.findByRole('radio', { name: expectedSecondSlot }),
-    );
-
-    // Go back and select initial slot
-    userEvent.click(screen.getByText(/^Prev/));
-    userEvent.click(screen.getByText(/^Prev/));
-
-    dayOfMonthButton = screen.getByLabelText(
-      new RegExp(slot308DateString, 'i'),
-    );
-    userEvent.click(dayOfMonthButton);
-    userEvent.click(await screen.findByRole('radio', { name: expected308 }));
-
-    // Have a selected slot, can move to next screen
-    userEvent.click(screen.getByText(/^Continue/));
-    await waitFor(() => {
-      expect(screen.history.push.called).to.be.true;
-    });
-    screen.history.push.reset();
-
-    // Clicking selected date should unselect date
-    userEvent.click(dayOfMonthButton);
-    userEvent.click(screen.getByText(/^Continue/));
-    await waitFor(() => {
-      expect(screen.getByRole('alert')).to.be.ok;
-    });
-    expect(screen.history.push.called).to.be.false;
+    MockDate.reset();
   });
+});
 
-  it('should show required text next to page heading', async () => {
-    const preferredDate = new Date();
-    const slot308Date = addDays(new Date(), 6);
-
-    setDateTimeSelectMockFetches({
-      typeOfCareId: 'outpatientMentalHealth',
-      slotDatesByClinicId: {
-        308: [slot308Date],
-      },
+describe('When preferred date is not immediate care', () => {
+  describe('And current date is last day of the month and end of day', () => {
+    beforeEach(() => {
+      mockFetch();
     });
 
-    const store = createTestStore(initialState);
-
-    await setTypeOfCare(store, /mental health/i);
-    await setVAFacility(store, '983', 'outpatientMentalHealth');
-    await setClinic(store, '983_308');
-    await setPreferredDate(store, preferredDate);
-
-    const screen = renderWithStoreAndRouter(
-      <Route component={DateTimeSelectPage} />,
-      {
-        store,
-      },
-    );
+    for (let i = 0; i < 12; i++) {
```

Comment:
I see why this was added during debugging but do we still need it?

Reply:
I think this is more than just debugging. These date edge cases have been a pain since the beginning.